app/javascript/lib/ajax_mixpanel_events.js

```diff
@@ -0,0 +1,38 @@
+import "../vendor/navigator.sendbeacon.min.js";
+
+const AjaxMixpanelEvents = (function () {
```

Comment:
There are a few things that bother me aesthetically about this code.

- It's using an immediately-invoked function expression in order to create an object with an `init` function, but it would be simpler in my opinion to have `AjaxMixpanelEvents` be a function (or perhaps `initAjaxMixpanelEvents` be a function).
- It calls `$` by assuming it's available as `window.$` but I'd prefer to see it do `import $ from jquery`.
- I'd prefer to see this code _not_ hook `$(document).ready`. In that case, whoever uses `AjaxMixpanelEvents` to be responsible for calling `$(document).ready(AjaxMixpanelEvents.init)`. That will make it easier to test this code if we ever want to add tests.

I know you inherited all of that before this pull request. :) None of these are blockers. Curious what you think.

Reply:
(This comment applies to most of the JS in this pull request.)

Reply:
Seems like Ben Sheldon gives a similar comment in https://github.com/codeforamerica/honeycrisp-gem/pull/241

> One file (or more) that has the implementation code and is not self-initializing and that could be used with Webpacker. And another file for backwards compatibility/sprockets that does the self-initializing.

Not a blocker, but curious for your take.

Reply:
i agree on the whole that I'd like to shake up how we're doing a lot of this stuff. I don't want to change too much right now because my goal is to have all of the JS work exactly like it did before the change to using webpacker so that we can move forward with some new feature work we'd like to do (ex. a tested autocomplete dropdown for filtering) without having to refactor all of the existing JS to do it. 

I'm hoping that we can kind of find time to refactor as we go along. For example, now that this can be testable code, it might make sense to test this code and refactor it to make it more testable as an eng task.